internal/access/credential.go

```diff
@@ -72,9 +72,24 @@ func UpdateCredential(c *gin.Context, user *models.Identity, newPassword string)
 	return nil
 }
 
-func updateCredential(c *gin.Context, user *models.Identity, newPassword string, isSelf bool) error {
+func updateCredential(c *gin.Context, user *models.Identity, oldPassword, newPassword string, isSelf, requireOldPassword bool) error {
 	db := getDB(c)
 
+	// Users have to supply their old password to change their existing password
+	if isSelf && requireOldPassword {
+		userCredential, err := data.GetCredential(db, data.ByIdentityID(user.ID))
+		if err != nil {
+			return fmt.Errorf("existing credential: %w", err)
+		}
+
+		// compare the stored hash of the user's password and the hash of the presented password
+		err = bcrypt.CompareHashAndPassword(userCredential.PasswordHash, []byte(oldPassword))
+		if err != nil {
+			// this probably means the password was wrong
```

Comment:
we should debug or trace level log the actual error here in case we need to debug this

Reply:
Added a trace level log.